pkg/controller/installation/core_controller.go

```diff
@@ -744,30 +752,46 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		reqLogger.Error(err, "An error occurred when querying the Installation resource")
 		return reconcile.Result{}, err
 	}
-	status := instance.Status
+
 	terminating := (instance.DeletionTimestamp != nil)
 	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
 
 	// Mark CR found so we can report converter problems via tigerastatus
 	r.status.OnCRFound()
 
+	// Changes for updating installation status conditions
+	if request.Name == InstallationName && request.Namespace == "" {
```

Comment:
If there are non-namespaced resources with the name "calico" this could introduce a bug. The only one I can think of would be if a user created a managed cluster called "calico". Would that be a deal breaker? Would it weigh up to the benefits of ending the reconciliation loop pre-emptively?

Since the reconcile.Request is created by one of our dependency, I do not think we can get richer information about what event triggered the reconcile loop.

That said, if we want to roll out these status conditions to other controllers, we must check per controller if the CR name clashes with any other resources. As far as I am aware Vara is going through this exercise  or is still in the process of going through this exercise.

Reply:
Maybe we can use another system to end the reconcile() function pre-emptively by relying on the `TigeraStatus.metadata.resourceVersion`. If we keep track either in memory or in the installation status if the TigeraStatus is new, in addition to checking the request name and request namespace, we can conclude that we can end the reconcile. If not, we continue the reconcile loop.

I considered proposing `TigeraStatus.metadata.generation`, but this value is always set to "1". It never increments, because we only update the `TigeraStatus.status`, not TigeraStatus itself.

Reply:
> I really don't like the possibility of a managed cluster named "calico" being a bug.

I agree.

> Would it be a problem if we did a full reconcile anytime TigeraStatus was updated?

Works for me.

Reply:
Done..